python/lsst/daf/persistence/registries.py

```diff
@@ -363,6 +368,37 @@ def lookup(self, lookupProperties, reference, dataId, **kwargs):
         cursor.execute(cmd, valueList)
         return [row for row in cursor.fetchall()]
 
+    def lookup(self, lookupProperties, reference, dataId, **kwargs):
+        """Perform a lookup in the registry.
+
+        Return values are refined by the values in dataId.
+        Returns a list of values that match keys in lookupProperties.
+        e.g. if the template is 'raw/raw_v%(visit)d_f%(filter)s.fits.gz', and
+        dataId={'visit':1}, and lookupProperties is ['filter'], and the
+        filesystem under self.root has exactly one file 'raw/raw_v1_fg.fits.gz'
+        then the return value will be [('g',)]
+
+        :param lookupProperties:
+        :param dataId: must be an iterable. Keys must be string.
+        If key is a string then will look for elements in the repository that match value for key.
+        If key is a 2-item iterable then will look for elements in the repository where the value is between
+        the values of key[0] and key[1].
+        :param reference: other data types that may be used to search for values.
+        :param **kwargs: nothing needed for sqlite lookup
+        :return: a list of values that match keys in lookupProperties.
+        """
+        if not self.conn:
+            return None
+
+        # input variable sanitization:
+        reference = sequencify(reference)
+        lookupProperties = sequencify(lookupProperties)
+
+        try:
+            return self._lookup(lookupProperties, dataId, reference)
+        except sqlite3.OperationalError:
```

Comment:
It's just a performance optimization not to have `checkColumns` always on?

Reply:
That's right.  I added some text to clarify this.